### ISSUE TITLE
[2.0.1]RuntimeException: Trying to call method appendmessage on a non-object

### ISSUE THREAD

**Issue by wojewsky** [phalcon/cphalcon#10405]
Hi,

with phalon 1.3.x we've added error messages to form elements by ->appendMessage.
With phalcon 2.0.1 i've got an runtime exception:
RuntimeException: Trying to call method appendmessage on a non-object

For example:

```php
    $form = new \Phalcon\Forms\Form();
    $element = new \Phalcon\Forms\Element\Text('field');
    $form->add($element);
    $element->appendMessage(new \Phalcon\Validation\Message('error message'));
    die(var_dump($form->getMessages()));
```

**Comment by noobiwan** [comment]
Any update on this? I'm using Phalcon 2.1.X and I'm getting the same error!

**Comment by noobiwan** [comment]
I was checking the source code and the problem seems to occur when `appendMessage()` is called before a call to `validate()`.  The method `appendMessage` depends on `this->_messages` being set ([line 346](https://github.com/phalcon/cphalcon/blob/2.1.x/phalcon/validation.zep#L346)) but it is only set in the `validate()` method at [line 111](https://github.com/phalcon/cphalcon/blob/2.1.x/phalcon/validation.zep#L111)

It seems to me that the logic is that `appendMessage` should be called in a *beforeValidation* event, that is triggered from the `validate()` method, but maybe there is a validity to allow messages to be added before that, or to throw an exception so the error is clear!

**Comment by Zaszczyk** [comment]
Up, I also have this problem.

**Comment by Knallcharge** [comment]
Same here. If I call "method_exists($validator, 'appendMessage')" it returns true, calling appendMessage on the next line will say that validator is not an object ... !?

**Comment by Jurigag** [comment]
@Knallcharge what phalcon version you are using ?

**Comment by Knallcharge** [comment]
@Jurigag 2.0.8

**Comment by Jurigag** [comment]
I have 2.0.13 and can't reproduce problem with code from OP. Can you post your code @Knallcharge to reproduce this ?

**Comment by Knallcharge** [comment]
Using the validator in a unit-test will result in the described behavior.

`$myValidator = new MyValidator();
$validation = new Phalcon\Validation();
$validation->bind(new stdClass(),
    array(
        'day'   => date('d'),
        'month' => date('m'),
        'year'  => date('Y')+1
    ));

$myValidator->validate($validation, '');`

printing out $validator in the validator itself will return it all right, but using "$validator->appendMessage" in the next line will tell me, it's not an object

**Comment by Jurigag** [comment]
Just wrote whole script to reproduce without necessary to have tests classes.

**Comment by sergeyklay** [comment]
Fixed in the `2.1.x` branch.
